app/license.go

```diff
@@ -46,46 +46,44 @@ func (a *App) LoadLicense() {
 }
 
 func (a *App) SaveLicense(licenseBytes []byte) (*model.License, *model.AppError) {
-	var license *model.License
-
-	if success, licenseStr := utils.ValidateLicense(licenseBytes); success {
-		license = model.LicenseFromJson(strings.NewReader(licenseStr))
+	success, licenseStr := utils.ValidateLicense(licenseBytes)
```

Comment:
Here I handled the "not success", and then, continue with the rest of the code. Instead thread the success as an else at the end.